File: src/main/java/com/ptah/common/exceptions/ApplicationException.java
package com.ptah.common.exceptions;

import camundajar.impl.scala.App;
import com.ptah.common.Errors;

public class ApplicationException extends CommonException{

    public ApplicationException(Errors error) {
        super(error.getMessage());
    }

    public static ApplicationException of(Errors error){
        return new ApplicationException(error);
    }

}

File: backend/src/main/kotlin/com/github/jarryzhou/ptah/common/exceptions/ApplicationException.kt
package com.github.jarryzhou.ptah.common.exceptions

import com.github.jarryzhou.ptah.common.Errors

class ApplicationException(error: Errors?) : CommonException(error?.message) {
    companion object {
        fun of(error: Errors?): ApplicationException = ApplicationException(error)
    }
}
